Arrowgene.Ddon.GameServer/Quests/Quest.cs

```diff
@@ -581,14 +583,40 @@ public virtual void PopulateStartingEnemyData(PartyQuestState partyQuestState)
                     continue;
                 }
 
-                foreach (var groupId in process.Blocks[(int)processState.BlockNo - 1].EnemyGroupIds)
+                foreach (var groupId in process.Blocks[processState.BlockNo - 1].EnemyGroupIds)
                 {
                     var enemyGroup = EnemyGroups[groupId];
                     partyQuestState.SetInstanceEnemies(this, enemyGroup.StageId, (ushort)enemyGroup.SubGroupId, enemyGroup.CreateNewInstance());
                 }
             }
         }
 
+        public virtual void HandleOmInstantValue(GameClient client, ulong key, uint value)
+        {
+            // Remove the valid bit (that way json doesn't need to provide it)
+            key = key & 0x7fffffffffffffff;
```

Comment:
What does this mean exactly? I think there should be a more in depth explanation

Reply:
Based on the dissassembly, this UID key value, always has the MSB set to 1 to mark that it is valid. So if the key is a value of 1, we would see the value here be 0x1000000000000001. So instead of forcing someone to provide a value with bit 63 set, either we can mask it off from the key or add it bad to the value while parsing the json.